### PR TITLE
#654 - AndInThreads - covered all ctors with tests.

### PR DESCRIPTION
**Pull request by driver733**
As per #654.
I have written unit tests for all of the constructors that were not tested in the `AndInThreads` class.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#687](https://github.com/yegor256/cactoos/issues/687) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#687](https://github.com/yegor256/cactoos/issues/687) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/510237597). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 1]
```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
         );
     }
 
+    @Test
+    public void testProcIterator() throws Exception {
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 can you change the test names from ``test...`` to ``worksWith...`` -- as in "class AndInThreads works with given Proc or somethnig else"

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 2]
```diff
@@ -46,6 +51,7 @@
  * @checkstyle JavadocMethodCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@SuppressWarnings("PMD.TooManyMethods")
 public final class AndInThreadsTest {
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 Please add javadocs over the tests, what is the input, what is the expected output and why? -- although I've seen that in general cactoos did not require test JavaDocs, I don't really understand how class ``AndInThreads`` is used on something else than Booleans. For instance, how does it work with a List of integers? It is not really clear from the test code alone.

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 3]
```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
...
+        );
+    }
+
+    @Test
+    public void testExecServiceProcIterable() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        final ExecutorService service = Executors.newSingleThreadExecutor();
+        new AndInThreads(
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 What is the returned ``value()`` here and why?

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 4]
```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
...
+        );
+    }
+
+    @Test
+    public void testExecServiceIteratorScalarBoolean() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                Executors.newSingleThreadExecutor(),
+                new ListOf<Scalar<Boolean>>(
+                    new Scalar.NoNulls<Boolean>(
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 Is it necessary to wrap the Constant boolean inside ``NotNulls``? Same for the other cases, I see it is in more places.

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 5]
```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
...
+                    ),
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(false)
+                    )
+                ).iterator()
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 Can you also add a test for empty collection? How does it behave if it has to conjunct an empty iterable?

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java [hunk 6]
```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
...
 }
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 I see multiple tests for logical conjunction between booleans, but only for ``true``/``false``. What about ``true``/``true``, ``false``/``false`` and ``false``/``true``?

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java (hunk 2 above)
Comment (driver733) [reply src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @amihaiemil Can I write failure explanation inside the tests instead?
I also did not get how this class works with various inputs from the first sight. I belive that the documentation of the AndInThreads class needs to be improved.

Comment (amihaiemil) [reply src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @driver733 

>Can I write failure explanation inside the tests instead?

Yes, that could also work. 

>I belive that the documentation of the AndInThreads class needs to be improved.

Then go ahead and open an Issue about it.

File: src/test/java/org/cactoos/scalar/AndInThreadsTest.java (hunk 3 above)
Comment (driver733) [reply src/test/java/org/cactoos/scalar/AndInThreadsTest.java]: @amihaiemil The result is always true. This has to be documented in the source class.
#688

**Comment by amihaiemil** [comment]
@driver733 looks good, thanks :)

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@driver733 can u change the title here to "Added tests for AndInThreads"? It's easier to search for this PR (I was just searching for it now and couldn't find it, it has a rather generic name)

**Comment by driver733** [comment]
@amihaiemil Done.

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+495](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "good": +20 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+890](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10604](https://www.0crat.com/u/yegor256)

**Comment by yegor256** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/687#issuecomment-367400351))

@yegor256 This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#687](https://github.com/yegor256/cactoos/issues/687) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%23687%27%7D) (restricted area)
